met-web/src/components/engagement/view/widgets/Timeline/TimelineWidgetView.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import { MetPaper, MetHeader2, MetParagraph, MetHeader4 } from 'components/common';
+import { Grid, Skeleton, Divider } from '@mui/material';
+import { Widget } from 'models/widget';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { TimelineWidget, TimelineEvent } from 'models/timelineWidget';
+import { fetchTimelineWidgets } from 'services/widgetService/TimelineService';
+
+interface TimelineWidgetProps {
+    widget: Widget;
+}
+
+const TimelineWidgetView = ({ widget }: TimelineWidgetProps) => {
+    const dispatch = useAppDispatch();
+    const [timelineWidget, setTimelineWidget] = useState<TimelineWidget>({
+        id: 0,
+        widget_id: 0,
+        engagement_id: 0,
+        title: '',
+        description: '',
+        events: [],
+    });
+    const [isLoading, setIsLoading] = useState(true);
+
+    const fetchTimeline = async () => {
+        try {
+            const timelines = await fetchTimelineWidgets(widget.id);
+            const timeline = timelines[timelines.length - 1];
+            setTimelineWidget(timeline);
+            setIsLoading(false);
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while fetching Engagement widgets information',
+                }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        fetchTimeline();
+    }, [widget]);
+
+    const handleRenderTimelineEvent = (tEvent: TimelineEvent, index: number) => {
+        const containerStylesObject = {
+            minHeight: index + 1 == timelineWidget.events.length ? '60px' : '80px',
+            display: 'flex',
+            flexDirection: 'row',
+            marginLeft: index + 1 == timelineWidget.events.length ? '24px' : '22px',
+            borderLeft: index + 1 == timelineWidget.events.length ? 'none' : '2px solid grey',
+        };
+        const circleContainerStylesObject = {
+            padding: '2px',
+            border: '2px solid grey',
+            marginLeft: '-25px',
+            display: 'inline-flex',
+            backgroundColor: '#fff',
+            borderRadius: '50%',
+            height: '48px',
+            width: '48px',
+        };
+        const circleStylesObject = {
+            width: '40px',
+            height: '40px',
+            borderRadius: '50%',
+            content: '""',
+            border: 2 == tEvent.status || 3 == tEvent.status ? '20px solid' : 'none',
```

Comment:
If there are only three statuses you could maybe save a bit of code with:

```suggestion
            border: 1 !== tEvent.status ? '20px solid' : 'none',
```

Also these equality operators should be using strict equality, i.e. ===